context/context.go

```diff
@@ -112,8 +112,15 @@ func NewPlmnSupportItem() (item factory.PlmnSupportItem) {
 }
 
 func (context *AMFContext) TmsiAllocate() int32 {
-	// val, err := AllocateUniqueID(&tmsiGenerator, "tmsi")
-	val, err := context.Drsm.AllocateInt32ID()
+	var val int32
+	var err error
+	if context.EnableDbStore {
+		val, err = context.Drsm.AllocateInt32ID()
+	} else {
+		var tmp int64
+		tmp, err = AllocateUniqueID(&tmsiGenerator, "tmsi")
+		val = int32(tmp)
```

Comment:
This type cast seems dangerous to me as it could potentially break if someone changes other areas of the code in the future. I think you can at least use the `ok` mechanism to catch errors here (please test it first):

```suggestion
		var tmp int64
		tmp, err = AllocateUniqueID(&tmsiGenerator, "tmsi")
		if val, ok := tmp.(int32); !ok {
		    // panic!
		}
```